xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -15155,6 +15155,7 @@ glusterd_check_brick_order(dict_t *dict, char *err_str, int32_t type,
         }
         free(pre_list);
         free(ai_list_tmp2);
+        ai_list_tmp2 = NULL;
```

Comment:
After being freed at the previous line and without being assigned a new value at 15167, the free at 15170 caused a segfault.
BTW, maybe I'm missing something, but aren't lines 15154 and 15167 pointless?

Reply:
Yes, it does not require to use ai_list_tmp2 at all at these places and freeing pre_list_tmp1 at the right time would be enough. But I don't know why it was done before so did not want to remove it and keep it as it is :)

Reply:
But maybe we should check it out. There might've been some logic before but it sort of feels like an unnecessary appendage now.